### DISCUSSION TITLE
Disable Serial output and input.

### DISCUSSION THREAD

**Discussion by AGRO666** [HomeSpan/HomeSpan#576] [Ideas]
Thank you for your excellent work. I have a suggestion. Sometimes in work it is required to use the serial port for other purposes. I suggest adding another log level. Thus, when the log level is set to homeSpan.setLogLevel(3), any output to Serial stops and processing of commands coming from Serial to HomeSpan is disabled.

**Comment by HomeSpan** [comment]
Thanks for the suggestion.  Something I will consider for potential future updates.  In the interim, have you tried using Serial1 or Serial2 instead (assuming your board has implemented these additional serial streams)?

**Reply by AGRO666** [reply]
I have no doubt that Serial 1 will work, but it is Serial that is used in my board, so I need to either redo the board or make changes to the library.

**Comment by tn-cs** [comment]
+1

homeSpan.setLogLevel(-1)

**Comment by HomeSpan** [comment]
Working on it now.  Will let you know when ready in the dev branch for testing.

**Comment by HomeSpan** [comment]
Code for this is now complete and can be tested in the dev branch.   If the HomeSpan Log Level is set to -1, all Serial OUTPUT is disabled.  You can set this programmatically in a sketch using `homeSpan.setLogLevel(-1)`, or by typing 'L -1' into the Serial Monitor.  

Serial INPUT is separately controlled with a new method `homeSpan.setSerialInputDisable(boolean disable)`.  Setting *disable=true* prevents HomeSpan from reading from the Serial port.  To resume normally operations, set *disable=false*.   You can use `homeSpan.getSerialInputDisable()` to retrieve the current *disable* status.

Please let me know if this does the job.  If so, I'll promote it to production in the next release.

**Reply by AGRO666** [reply]
I have tested both functions. `homeSpan.setSerialInputDisable(1)` disables the input as well. But `homeSpan.setLogLevel(-1)` in Serial I see *** HOMESPUN STATUS CHANGE: WiFi Connecting and *** HOME SPAM STATUS CHANGE: Device Paired

**Comment by HomeSpan** [comment]
Setting the Log Level to -1 only suppresses messages output by the HomeSpan library, not anything that you might include in the sketch.  I'm guessing the messages above (HOME STATUS CHANGE...) are from the sketch code itself, likely based on an example of how to use the `homeSpan.setStatusCallback()` method.  That example uses Serial.printf().  You would need to switch those to LOG0() in order to have them controlled by the HomeSpan Log Level.

Note setting the Log Level to -1 only prevents messages produced with the LOG() macro from being output to the Serial Monitor.  It does not disable the entire Serial port so you can still use it for other things.  I'll update to the documentation to make this clearer.

**Reply by AGRO666** [reply]
Sorry, my mistake, this is really the output from the sample code. Both procedures work perfectly. Great job.

**Comment by HomeSpan** [comment]
Excellent - I'l promote to production in next release.
